### PR TITLE
WT-7428 Move bstorage usage out of btree and into tiered structure.

### PR DESCRIPTION
**Pull request by sueloverso**

### CODE REVIEW

File: src/tiered/tiered_config.c [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+/*
+ * __wt_tiered_config_bstorage --
+ *     Return a bucket storage handle based on the configuration.
+ */
+int
+__wt_tiered_config_bstorage(
+  WT_SESSION_IMPL *session, const char **cfg, WT_BUCKET_STORAGE **bstoragep)
+{
+    WT_BUCKET_STORAGE *bstorage;
```
Comment (keitharnoldsmith) [inline src/tiered/tiered_config.c:201]: Unused variable? (It's set at the end of the function, but never used.)

**Review by keitharnoldsmith — APPROVED** [review]
LGTM.  A couple of comments on small fixes/changes.  It looks like the unused variable is what is causing the Evergreen compile to fail.

File: src/schema/schema_create.c [hunk 2]
```diff
@@ -845,6 +860,8 @@ __create_tiered(WT_SESSION_IMPL *session, const char *uri, bool exclusive, const
 
 err:
     WT_TRET(__wt_schema_release_tiered(session, &tiered));
+    if (tmp_free)
+        __wt_scr_free(session, &tmp);
```
[the remark this answers is not in the document]
Comment (sueloverso) [reply src/schema/schema_create.c:864]: Cleaned up. It was already `WT_DECL_ITEM`.

File: src/tiered/tiered_config.c (hunk 1 above)
Comment (sueloverso) [reply src/tiered/tiered_config.c:201]: Good catch! Thanks. There used to be more there that got moved/changed... Fixed.

**Review by ddanderson — APPROVED** [review]
LGTM!

**Comment by sueloverso** [comment]
@keitharnoldsmith and @ddanderson at the moment the one failure is a cache stuck in {{test_rollback_to_stable10.py}}. That is a known, occasional failure. Assuming all other tests pass I will merge.

**Comment by keitharnoldsmith** [comment]
Works for me. 👍

**Comment by ddanderson** [comment]
Sounds good - you can retry just that one failure if you want to see it green :-)

**Pull request merged**